### DISCUSSION TITLE
Nonebot是否考虑给插件和适配器之间添加中间层

### DISCUSSION THREAD

**Discussion by boxie123** [nonebot/discussions#448] [想法]
目前各个插件的消息收发和适配器高度耦合，在之前普遍使用`onebot`协议的情况下问题并不明显。

但在当前环境下，各种适配器如`red`、`satori`等等逐渐增多，跨平台收发消息的需求也一直存在，如[`plugin-alconna`](https://github.com/nonebot/plugin-alconna)、[`nonebot-plugin-send-anything-anywhere`](https://github.com/MountainDash/nonebot-plugin-send-anything-anywhere)等插件虽然可以实现一些功能，但并无统一标准，给插件编写、适配增加了很多难度。

想请问`Nonebot`本身有没有计划给插件和适配器之间添加中间层，使多平台多协议适配更加简单？

**Comment by yanyongyu** [comment]
nb本体不会介入平台交互（因为本身兼容各类平台的中间层就是很💩的），你可以直接使用alconna之类的插件，在文档->最佳实践内也有对跨平台处理的教程。

**Comment by Lancercmd** [comment]
你可以考虑把nb作为中间层，nb是一个搭载平台适配器的消息处理器。
